sdcm/utils/decorators.py

```diff
@@ -333,3 +333,30 @@ def wrapper(*args, **kwargs):
                 raise UnsupportedNemesis("Capacity Issue") from ex
             raise
     return wrapper
+
+
+def optional_stage(stage_names: str | list[str]):
+    """
+    Decorator checks if the provided test stage is enabled in the test configuration
+    'optional_test_stages' parameter. If the stage is disabled, the decorated function is skipped.
+    More details can be found in https://github.com/scylladb/scylla-cluster-tests/blob/master/docs/faq.md
+
+    :param stage_names: str or list, name of the test stage(s)
+    """
+    stage_names = stage_names if isinstance(stage_names, list) else [stage_names]
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            # making import here, to work around circular import issue
+            from sdcm.cluster import TestConfig
+            optional_test_stages = TestConfig().tester_obj().optional_test_stages
+            skipped_stages = [stage for stage in stage_names if not optional_test_stages[stage]]
+
+            if not skipped_stages:
+                return func(*args, **kwargs)
+            else:
+                skipped_stages_str = ', '.join(skipped_stages)
+                LOGGER.info(f"'{func.__name__}' is skipped as '{skipped_stages_str}' test stage(s) is disabled.")
```

Comment:
Maybe warning, so it would be visible in stdout as well ? i.e. on Jenkins output ?

Reply:
Agree. Changed this, now it is:
```
< t:2024-09-30 18:31:20,143 f:decorators.py   l:360  c:sdcm.utils.decorators p:WARNING > 'run_pre_create_keyspace' is skipped as 'prepare_write' test stage(s) is disabled.
< t:2024-09-30 18:31:20,143 f:decorators.py   l:360  c:sdcm.utils.decorators p:WARNING > 'run_pre_create_schema' is skipped as 'prepare_write' test stage(s) is disabled.
```